### PR TITLE
Add tests for "_adb_EdgeRequestWorker"

### PR DESCRIPTION
**Pull request by yangyansong-adbe**

### CODE REVIEW

File: code/AdobeEdge.brs [hunk 1]
```diff
@@ -931,19 +932,39 @@ function _adb_buildEdgeRequestURL(configId as string, requestId as string, edgeD
...
         _queue_size_max: 100,
 
-        queue: function(requestId as string, xdmData as object, timestamp as integer, configId = invalid as dynamic, ecid = invalid as dynamic, edgeDomain = invalid as dynamic) as void
+        queue: function(requestId as string, xdmData as object, timestamp as integer) as void
+            if _adb_isEmptyOrInvalidString(requestId)
+                _adb_log_debug("[EdgeRequestWorker.queue()] requestId is invalid")
+                return
+            end if
+
+            if _adb_isEmptyOrInvalidObject(xdmData)
```
Comment (addb) [inline code/AdobeEdge.brs:954]: let's call it isEmptyOrInvalidMap. I added several helpers with similar name.

File: code/AdobeEdge.brs [hunk 2]
```diff
@@ -931,19 +932,39 @@ function _adb_buildEdgeRequestURL(configId as string, requestId as string, edgeD
 end function
 
 function _adb_EdgeRequestWorker(stateManager as object) as object
+    if stateManager = invalid
+        _adb_log_debug("stateManager is invalid")
+        return invalid
+    end if
+    ' if stateManager._type_ = invalid or stateManager._type_ <> "stateManager"
+    '     _adb_log_debug("not a StateManager object")
```
Comment (addb) [inline code/AdobeEdge.brs:940]: Should we remove this? If this is an improvement let's open a task issue for this.

Comment (yangyansong-adbe) [reply code/AdobeEdge.brs]: Removed.

File: code/AdobeEdge.brs [hunk 3]
```diff
@@ -931,19 +932,35 @@ function _adb_buildEdgeRequestURL(configId as string, requestId as string, edgeD
 end function
 
 function _adb_EdgeRequestWorker(stateManager as object) as object
+    if stateManager = invalid
```
Comment (addb) [inline code/AdobeEdge.brs:935]: Why are we getting stateManager as parameter and not `_adb_stateManager()`?

File: code/AdobeEdge.brs [hunk 4]
```diff
@@ -570,15 +570,10 @@ function _adb_task_node_EventProcessor(internalConstants as object, task as obje
...
-            ecid = m.stateManager.getECID()
-            configId = m.stateManager.getConfigId()
-            edgeDomain = m.stateManager.getEdgeDomain()
-
-
             requestId = event.uuid
             xdmData = event.data
 
-            m.edgeRequestWorker.queue(requestId, xdmData, event.timestamp_in_millis, configId, ecid, edgeDomain)
+            m.edgeRequestWorker.queue(requestId, xdmData, event.timestamp_in_millis)
```
Comment (addb) [inline code/AdobeEdge.brs:576]: if everything we want is coming from event, why not just pass the event and extract requesstID, xdmData, ts inside the queue.

File: code/AdobeEdge.brs [hunk 5]
```diff
@@ -970,19 +965,35 @@ function _adb_buildEdgeRequestURL(configId as string, requestId as string, edgeD
...
+        return invalid
+    end if
     instance = {
         _queue: [],
         _stateManager: stateManager
         _queue_size_max: 100,
 
-        queue: function(requestId as string, xdmData as object, timestamp as integer, configId = invalid as dynamic, ecid = invalid as dynamic, edgeDomain = invalid as dynamic) as void
+        queue: function(requestId as string, xdmData as object, timestamp as integer) as void
+            if _adb_isEmptyOrInvalidString(requestId)
```
Comment (addb) [inline code/AdobeEdge.brs:978]: we will never have empty or invalid request id since we internally generate it.

File: code/AdobeEdge.brs [hunk 6]
```diff
@@ -970,19 +965,35 @@ function _adb_buildEdgeRequestURL(configId as string, requestId as string, edgeD
...
+                _adb_log_debug("[EdgeRequestWorker.queue()] requestId is invalid")
+                return
+            end if
+
+            if isEmptyOrInvalidMap(xdmData)
```
Comment (addb) [inline code/AdobeEdge.brs:983]: We already check this in the _sendEvent() function

File: code/AdobeEdge.brs [hunk 7]
```diff
@@ -970,19 +965,35 @@ function _adb_buildEdgeRequestURL(configId as string, requestId as string, edgeD
...
+                _adb_log_debug("[EdgeRequestWorker.queue()] xdmData is invalid")
+                return
+            end if
+
+            if timestamp <= 0
```
Comment (addb) [inline code/AdobeEdge.brs:988]: Again, this should not be the case since this is generated by us.

File: code/AdobeEdge.brs [hunk 8]
```diff
@@ -1075,7 +1074,9 @@ function _adb_EdgeRequestWorker(stateManager as object) as object
             url = _adb_buildEdgeRequestURL(configId, requestId, edgeDomain)
             _adb_log_verbose("request JSON: " + FormatJson(jsonBody))
             response = _adb_serviceProvider().networkService.syncPostRequest(url, jsonBody)
-            response.requestId = requestId
```
Comment (addb) [inline code/AdobeEdge.brs:1078]: Let's add a helper to generate identity map. That helper should have access to statemanager.

File: code/AdobeEdge.brs (hunk 4 above)
Comment (yangyansong-adbe) [reply code/AdobeEdge.brs:576]: If so, `RequestWorker` has to know the event object structure. It should be clearer to keep `Event` operations inside the event processor.

File: code/AdobeEdge.brs (hunk 8 above)
Comment (yangyansong-adbe) [reply code/AdobeEdge.brs:1078]: It's not a good time to refactor it now, we can do it when supporting other `identities` in the xdm request.

File: code/AdobeEdge.brs [hunk 9]
```diff
@@ -570,15 +570,10 @@ function _adb_task_node_EventProcessor(internalConstants as object, task as obje
                 return
```
Comment (addb) [inline code/AdobeEdge.brs:570]: We can move this check  if (not m._hasValidConfig())

to processRequests() where we check for valid ecid and configid

File: code/AdobeEdge.brs (hunk 8 above)
Comment (addb) [reply code/AdobeEdge.brs:1078]: Agreed

File: code/AdobeEdge.brs (hunk 4 above)
Comment (addb) [reply code/AdobeEdge.brs:576]: sounds good.

File: code/AdobeEdge.brs (hunk 5 above)
Comment (yangyansong-adbe) [reply code/AdobeEdge.brs:978]: Discussed offline, keep it for now.

**Review by addb — APPROVED** [review]

**Pull request merged**